### PR TITLE
kmsv2: perform transformation and decoding in parallel during storage list calls

### PR DESCRIPTION
**Pull request by enj**
/kind feature

xref: #88633

```release-note
TODO
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go [hunk 1]
```diff
@@ -748,27 +752,87 @@ func (s *store) GetList(ctx context.Context, key string, opts storage.ListOption
 			growSlice(v, 2048, len(getResp.Kvs))
 		}
 
+		// needs to be large enough to give value in terms of parallelizing work
```
Comment (smarterclayton) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go:755]: Changing all Lists to use at minimum one extra goroutine feels undesirable.

Also, this is a lot of code to add to a function.  Is there an easier to read way to factor this?

Comment (smarterclayton) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go:755]: There’s also minimal value in this when a kms isn’t in use?  Non kms clusters get slower / have more gosched pressure?

Any benchmarks showing how this impacts very large lists both with and without kms (benchmark is probably required for this to merge with a variety of factors considered)?

**Comment by furkatgofurov7** [comment]
Hi @enj!

K8s 1.27 Bug Triage Shadow here. FYI, with Code Freeze approaching next week on March 15th, if you would like this PR to be part of the v1.27 release, please make sure it goes in before above mentioned date. Post Code Freeze, we will only be able to accept Release Blockers and Critical issues. Thank you!

**Comment by enj** [comment]
With #116155 merged this PR is no longer necessary.  KMS v2 no longer has any significant IO bound code as the number of DEKs will always be small (i.e. remote KMS calls are minimal now).  Performing encryption/decryption in parallel does not really help because the CPU can only do so much math at one time.

**Pull request closed without merging**